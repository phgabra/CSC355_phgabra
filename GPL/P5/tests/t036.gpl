// explicitly initialize a string to empty
string empty_string = "";
string hello = "hello";
string world = "world";
string space_string = " ";

string aa = "hello" + empty_string + "world";
string ab = "hello" + "" + "world";
string ac = hello + empty_string + world;

string ad = hello + empty_string + "world";
string ae = hello + "" + world;
string af = hello + empty_string + world;

string ag = hello + space_string + "world";
string ah = hello + " " + world;
string ai = hello + space_string + world;
