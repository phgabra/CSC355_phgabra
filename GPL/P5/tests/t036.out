gpl.cpp::main()
  input file(tests/t036.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(false)
  symbol_table(true)
  graphics(false)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().

No errors found (parser probably worked correctly).


Printing the symbol table from main()
string aa = "helloworld"
string ab = "helloworld"
string ac = "helloworld"
string ad = "helloworld"
string ae = "helloworld"
string af = "helloworld"
string ag = "hello world"
string ah = "hello world"
string ai = "hello world"
string empty_string = ""
string hello = "hello"
string space_string = " "
string world = "world"
Graphics is turned off by the Makefile.  Program exiting.

gpl.cpp::main() done.
